aquarius/events/decryptor.py

```diff
@@ -17,18 +19,34 @@
 
 def decrypt_ddo(w3, provider_url, contract_address, chain_id, txid, hash, es_instance):
     aquarius_account = get_aquarius_wallet()
-    nonce = str(int(datetime.now(timezone.utc).timestamp()))
+    # get nonce
+    try:
+        params = {"userAddress": aquarius_account.address}
+        nonce_response = requests.get(
+            provider_url + "/api/services/nonce",
+            params=params,
+            timeout=4,
+        ).json()
+
+        if nonce_response:
```

Comment:
I think you can remove this if statement to trigger the exception handling itself. Otherwise, assuming you don't have a response, it will not fall back because it doesn't throw an exception. Instead, it will exit the if, and go to line 40, where the nonce will be undefined.